tests/core_tests/test_log.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import copy
+
+import sgtk
+from tank_test.tank_test_base import TankTestBase
+
+
+class TestLogManager(TankTestBase):
+    """Tests the LogManager interface."""
+    def test_global_debug_environment(self):
+        """
+        Ensures that the debug logging environment variable is set/unset when
+        the global debug logging property is toggled.
+        """
+        manager = sgtk.log.LogManager()
+        original_env = copy.copy(os.environ)
+        original_debug = manager.global_debug
+        debug_name = sgtk.constants.DEBUG_LOGGING_ENV_VAR
+
+        try:
+            if debug_name in os.environ:
+                del os.environ[debug_name]
+
+            manager.global_debug = True
+            self.assertTrue((debug_name in os.environ))
```

Comment:
You could also do `self.assertIn(debug_name, os.environ)`
I wish we used pytest, we would only be using assert debug_name in os.environ and pytest would instrument the code properly on import to say something like
`Assertion Failed, <somevalue> not found in os.environ`